pkg/auth/providers/common/usermanager.go

```diff
@@ -303,6 +313,58 @@ func (m *userManager) EnsureUser(principalName, displayName string) (*v3.User, e
 	return user, nil
 }
 
+func (m *userManager) CreateNewUserClusterRoleBinding(userName string, userUID apitypes.UID) error {
+	roleName := userName + "-view"
+	bindingName := "grb-" + roleName
+
+	ownerReference := v1.OwnerReference{
+		APIVersion: "management.cattle.io/v3",
+		Kind:       "User",
+		Name:       userName,
+		UID:        userUID,
+	}
+
+	rule := k8srbacv1.PolicyRule{
+		Verbs:         []string{"get"},
+		APIGroups:     []string{"management.cattle.io"},
+		Resources:     []string{"users"},
+		ResourceNames: []string{userName},
+	}
+	role := &k8srbacv1.ClusterRole{
+		ObjectMeta: v1.ObjectMeta{
+			Name:            roleName,
+			OwnerReferences: []v1.OwnerReference{ownerReference},
+		},
+		Rules: []k8srbacv1.PolicyRule{rule},
+	}
+
+	cr, err := m.RBAC.ClusterRoles("").Create(role)
+	if err != nil {
```

Comment:
i think we need to do a check if the role exists and we need to not error out if the error is "is already exists"